--- configure.h.in	2006-08-04 12:47:33.000000000 -0500
+++ configure.h.in	2006-10-18 18:42:57.000000000 -0500
@@ -108,9 +108,9 @@
 #define KDE_APPS_PATH		"$KDEDIR/share/applnk:$KDEDIR/share/applications:/usr/share/applnk:/usr/share/applications"
 #define KDE_ICONS_PATH		"~/.icons/kde/48x48/apps/:$XDG_DATA_DIRS/icons/kde/48x48/apps/:$KDEDIR/share/icons/default.kde/48x48/apps:$KDEDIR/share/icons/hicolor/48x48/apps:$KDEDIR/share/icons/locolor/48x48/apps:/usr/share/pixmaps"
 #define KDE_CACHE_FILE		AFTER_NONCF "/KDECategories"
-#define SYSTEM_APPS_PATH     	"/etc/X11/applnk:/usr/share/applications:/usr/local/share/applications"
-#define SYSTEM_ICONS_PATH    	"/usr/share/pixmaps:/usr/local/share/pixmaps:/usr/share/icons:/usr/share/icons/hicolor/48x48/apps"
-#define SYSTEM_CACHE_FILE	AFTER_NONCF "/SystemCategories"
+#define OTHER_APPS_PATH     "/etc/X11/applnk:/usr/share/applications:/usr/local/share/applications"
+#define OTHER_ICONS_PATH    "/usr/share/pixmaps:/usr/local/share/pixmaps:/usr/share/icons:/usr/share/icons/hicolor/48x48/apps"
+#define OTHER_CACHE_FILE	AFTER_NONCF "/OtherCategories"
 
 
 #define DEFAULT_CAPTURE_WINDOW_FILE		"~/capture_window"
