--- src/afterstep/configure.c	2006-09-26 18:45:18.000000000 -0500
+++ src/afterstep/configure.c	2007-02-23 13:12:13.000000000 -0600
@@ -1697,6 +1697,7 @@
             }
         	if( UpdateGtkRC(Environment) ) 
 				ExecuteFunction (&gtkrc_signal_func, &dummy_event, -1);
+			if( !get_flags(Environment->flags,ASE_NoKDEGlobalsTheming ) )
 			if( UpdateKCSRC() ) 
 				ExecuteFunction (&kde_signal_func, &dummy_event, -1);
 		}
@@ -1785,6 +1786,7 @@
         what = PARSE_EVERYTHING ;
 		if( UpdateGtkRC(Environment) ) 
 			ExecuteFunction (&gtkrc_signal_func, &dummy_event, -1);
+		if( !get_flags(Environment->flags,ASE_NoKDEGlobalsTheming ) )
 		UpdateKCSRC();
     }
 
