--- libAfterImage/apps/asview.c	2005-06-08 09:16:27.000000000 -0500
+++ libAfterImage/apps/asview.c	2006-10-12 14:04:36.000000000 -0500
@@ -84,7 +84,7 @@
 	depth = DefaultDepth( dpy, screen );
 	/* see ASView.3 : */
 	asv = create_asvisual( dpy, screen, depth, NULL );
-	/*asv = create_asvisual_for_id( dpy, screen, depth, 0x25, None, NULL ); */
+	/* asv = create_asvisual_for_id( dpy, screen, depth, 0x28, None, NULL ); */
 #endif	
 	/* see ASView.2 : */
 	im = file2ASImage( image_file, 0xFFFFFFFF, SCREEN_GAMMA, 0, getenv("IMAGE_PATH"), NULL );
