--- libAfterStep/asapp.c	2006-09-21 17:56:32.000000000 -0500
+++ libAfterStep/asapp.c	2006-10-18 18:42:57.000000000 -0500
@@ -98,7 +98,7 @@
 struct ASCategoryTree *AfterStepCategories = NULL ;
 struct ASCategoryTree *KDECategories = NULL ;
 struct ASCategoryTree *GNOMECategories = NULL ;
-struct ASCategoryTree *SystemCategories = NULL ;
+struct ASCategoryTree *OtherCategories = NULL ;
 struct ASCategoryTree *CombinedCategories = NULL ;
 
 
@@ -891,9 +891,9 @@
 	{	
 		ct = GNOMECategories ; 
 		offset = 6 ; 
-	}else if( !mystrncasecmp (name, "SYSTEM:", 7) )
+	}else if( !mystrncasecmp (name, "OTHER:", 7) )
 	{	
-		ct = SystemCategories ;
+		ct = OtherCategories ;
 		offset = 7 ; 
 	}else if( !mystrncasecmp (name, "COMBINED:", 9) )
 	{	
