--- AfterStep.desktop.in	2013-05-01 08:34:11.000000000 -0500
+++ AfterStep.desktop.in	2013-12-26 22:11:28.000000000 -0600
@@ -1,5 +1,5 @@
-[Window Manager]
-SessionManaged=true
+#[Window Manager]
+#SessionManaged=true
 
 [Desktop Entry]
 Name=AfterStep
@@ -7,7 +7,7 @@
 Comment=AfterStep is feature rich NeXTish window manager.
 # name we put on the WM spec check window
 X-GNOME-WMName=AfterStep
-# back compat only 
+# back compat only
 X-GNOME-Autostart-Phase=WindowManager
 X-GNOME-Provides=windowmanager
 X-GNOME-Autostart-Notify=true
