--- src/WinTabs/WinTabs.c	2006-09-21 17:56:32.000000000 -0500
+++ src/WinTabs/WinTabs.c	2006-10-13 17:14:26.000000000 -0500
@@ -1095,6 +1095,7 @@
 	}
     LOCAL_DEBUG_OUT( "creating main window with geometry %dx%d%+d%+d", width, height, x, y );
     w = create_visual_window( Scr.asv, Scr.Root, x, y, 1, 1, 0, InputOutput, CWBackPixmap, &attributes);
+    LOCAL_DEBUG_OUT( "main window created with Id %lX", w);
     set_client_names( w, Config->title?Config->title:"WinTabs", 
 						 Config->icon_title?Config->icon_title:"WinTabs - iconic", 
 					  	 CLASS_WINTABS, MyName );
