--- configure.h.in	2013-10-06 22:53:46.637032758 -0500
+++ configure.h.in	2013-05-01 08:34:11.000000000 -0500
@@ -106,7 +106,7 @@
 #define STANDARD_CATEGORIES_FILE 	"standard_categories" 
 #define AFTERSTEP_APPS_DIR		"applications" 
 #define AFTERSTEP_CACHE_FILE	AFTER_NONCF "/AfterStepCategories"
-#define GNOME_APPS_PATH		"~/.gnome2/vfolders/applications/:$GNOMEDIR/share/applications:/usr/share/applications:/usr/local/share/applications"
+#define GNOME_APPS_PATH		"~/.gnome2/vfolders/applications/:$GNOMEDIR/share/applications:/usr/share/applications"
 #define GNOME_ICONS_PATH	"~/.icons/gnome/48x48/apps/:$XDG_DATA_DIRS/icons/gnome/48x48/apps/:$GNOMEDIR/share/pixmaps:/usr/share/pixmaps:/usr/share/icons:/usr/share/icons/hicolor/48x48/apps:/usr/share/icons/hicolor/32x32/apps:/usr/share/icons/gnome/48x48/apps"
 #define GNOME_CACHE_FILE	AFTER_NONCF "/GNOMECategories"
 #define KDE_APPS_PATH		"$KDEDIR/share/applnk:$KDEDIR/share/applications:/usr/share/applnk:/usr/share/applications"
