--- src/ascp/asimbrowser.c	2005-10-05 15:46:23.000000000 -0500
+++ src/ascp/asimbrowser.c	2009-11-01 21:39:37.000000000 -0600
@@ -92,7 +92,7 @@
 	int i;
 
 	/* Save our program name - for error messages */
-    InitMyApp (CLASS_ASIMBROWSER, argc, argv, asimbrowser_usage, NULL, 0 );
+    InitMyApp (CLASS_ASIMBROWSER, argc, argv, NULL, asimbrowser_usage, 0 );
 
 	memset( &ASIMBrowserState, 0x00, sizeof(ASIMBrowserState ));
 	ASIMBrowserState.html_save = default_html_save;
