--- src/ASDocGen/ASDocGen.c	2009-02-11 17:32:03.000000000 -0600
+++ src/ASDocGen/ASDocGen.c	2009-11-01 21:36:25.000000000 -0600
@@ -218,7 +218,7 @@
 	ASDocType target_type = DocType_Source ;
 	/* Save our program name - for error messages */
 	set_DeadPipe_handler(DeadPipe);
-    InitMyApp (CLASS_ASDOCGEN, argc, argv, asdocgen_usage, NULL, 0 );
+    InitMyApp (CLASS_ASDOCGEN, argc, argv, NULL, asdocgen_usage, 0 );
 	LinkAfterStepConfig();
 	InitSession();
     for( i = 1 ; i< argc ; ++i)
