--- src/afterstep/configure.c	2009-04-01 16:35:09.000000000 -0500
+++ src/afterstep/configure.c	2009-11-01 21:43:07.000000000 -0600
@@ -359,6 +359,7 @@
 	{"WinListSortOrder", SetInts, (char**)&TmpFeel.winlist_sort_order, (int *)&dummy},
 	{"WinListHideIcons", SetFlag2, (char**)WinListHideIcons, NULL},
 	{"SuppressIcons", SetFlag2, (char**)SuppressIcons, NULL},
+	{"WarpPointer", SetFlag2, (char**)WarpPointer, NULL},
 
     /* look options */
 	/* obsolete stuff */
