--- src/ascp/ascolor.c	2009-04-03 09:15:49.000000000 -0500
+++ src/ascp/ascolor.c	2009-11-01 21:40:01.000000000 -0600
@@ -231,7 +231,7 @@
 	int i;
 
 	/* Save our program name - for error messages */
-    InitMyApp (CLASS_ASCOLOR, argc, argv, ascolor_usage, NULL, 0 );
+    InitMyApp (CLASS_ASCOLOR, argc, argv, NULL, ascolor_usage, 0 );
 
 	memset( &ASColorState, 0x00, sizeof(ASColorState ));
 	ASColorState.doc_str = default_doc_str;
