--- libAfterStep/asapp.h	2006-09-19 13:13:47.000000000 -0500
+++ libAfterStep/asapp.h	2006-10-18 18:42:57.000000000 -0500
@@ -342,7 +342,7 @@
 extern struct ASCategoryTree *AfterStepCategories ;
 extern struct ASCategoryTree *KDECategories ;
 extern struct ASCategoryTree *GNOMECategories ;
-extern struct ASCategoryTree *SystemCategories ;
+extern struct ASCategoryTree *OtherCategories ;
 extern struct ASCategoryTree *CombinedCategories ;
 
 /* this two are unused in as-stable yet : */
