--- libASGTK/asgtklookedit.c	2008-07-02 09:46:01.000000000 -0500
+++ libASGTK/asgtklookedit.c	2009-08-01 15:42:43.000000000 -0500
@@ -25,6 +25,7 @@
 #include "../libAfterImage/asvisual.h"
 #include "../libAfterStep/asapp.h"
 #include "../libAfterStep/screen.h"
+#include "../libAfterStep/session.h"
 #include "../libAfterStep/parser.h"
 #include "../libAfterStep/freestor.h"
 #include "../libAfterStep/mystyle.h"
