--- src/state.c	2010-03-09 12:38:08.000000000 -0600
+++ src/state.c	2013-11-21 08:58:39.246582035 -0600
@@ -191,7 +191,8 @@
     if(vf->ptr >= size)
 	return 0;
     
-    nact = AV_MIN(nbyte, (avsize_t) (size - vf->ptr));
+    // since nbyte is avsize_t, the min will not be larger than that datatype
+    nact = AV_MIN((avoff_t)nbyte, (avoff_t) (size - vf->ptr));
     
     memcpy(buf, sf->contents + vf->ptr, nact);
     
