--- src/archive.c	2011-06-29 15:20:31.000000000 -0500
+++ src/archive.c	2013-11-21 08:59:03.578330532 -0600
@@ -426,7 +426,7 @@
     avoff_t realoff;
     struct archfile *fil = arch_vfile_file(vf);
     struct archnode *nod = fil->nod;
-    avsize_t nact;
+    avoff_t nact;
 
     if(AV_ISDIR(nod->st.mode))
         return -EISDIR;
@@ -435,9 +435,10 @@
         return 0;
 
     realoff = vf->ptr + nod->offset;
-    nact = AV_MIN(nbyte, (avsize_t) (nod->realsize - vf->ptr));
+    nact = AV_MIN((avoff_t)nbyte, (avoff_t) (nod->realsize - vf->ptr));
 
-    res = av_pread(fil->basefile, buf, nact, realoff);
+    // due to the MIN, nact is not larger than the range of avsize_t
+    res = av_pread(fil->basefile, buf, (avsize_t)nact, realoff);
     if(res > 0)
         vf->ptr += res;
 
