--- Makefile	2009-10-27 03:07:47.000000000 -0500
+++ Makefile	2009-10-27 20:39:26.000000000 -0500
@@ -41,16 +41,6 @@
 	cp -R data/* $(DESTDIR)$(DATADIR)
 	cp $(PROGRAM) $(DESTDIR)$(BINDIR)
 	cp -R locales/* $(DESTDIR)$(DATADIR)/locales
-	#  install .desktop file (menu entry)
-	@echo "[Desktop Entry]" > $(DESKTOP)
-	@echo "Name=$(PROGRAM)" >> $(DESKTOP)
-	@echo "Categories=Graphics" >> $(DESKTOP)
-	@echo "Type=Application" >> $(DESKTOP)
-	@echo "Terminal=false" >> $(DESKTOP)
-	@echo "Exec=$(BINDIR)/$(PROGRAM)" >> $(DESKTOP)
-	@echo "Icon=$(DATADIR)/icons/$(PROGRAM).png" >> $(DESKTOP)
-	@echo "GenericName=Digital Photo Editor" >> $(DESKTOP)
-	xdg-desktop-menu install $(DESKTOP)
 	#  install man page
 	cp -f doc/$(PROGRAM).man $(PROGRAM).1
 	gzip -f -9 $(PROGRAM).1
