--- build	2008-05-21 10:22:52.000000000 -0500
+++ build	2008-05-22 11:27:18.000000000 -0500
@@ -38,32 +38,9 @@
 
 PREFIX="$DESTDIR$PREFIX"
 
-if [ -n "$PREFIX" ]; then
-   BINDIR="$PREFIX/bin"
-   DATADIR="$PREFIX/share/$APPNAME"
-   DOCDIR="$PREFIX/share/doc/$APPNAME"
-else
-   if [ $(id -u) -eq 0 ]; then  PREFIX=/usr/local
-   else  PREFIX="$HOME"; fi
-fi
-
-# if not predefined, query install locations
-
-if [ -z "$BINDIR" ]; then 
-   BINDIR="$PREFIX/bin"
-   read -p "binary directory: [ $BINDIR ] " dir
-   if [ -n "$dir" ]; then BINDIR="$dir"; fi
-fi
-if [ -z "$DATADIR" ]; then 
-   DATADIR="$PREFIX/share/$APPNAME"
-   read -p "data directory: [ $DATADIR ] " dir
-   if [ -n "$dir" ]; then DATADIR="$dir"; fi
-fi
-if [ -z "$DOCDIR" ]; then 
-   DOCDIR="$PREFIX/share/doc/$APPNAME"
-   read -p "doc directory: [ $DOCDIR ] " dir
-   if [ -n "$dir" ]; then DOCDIR="$dir"; fi
-fi
+BINDIR="/usr/bin"
+DATADIR="/usr/share/$APPNAME"
+DOCDIR="/usr/share/doc/$APPNAME-0.41"
 
 # check prerequisites: g++ and libgtk2.0-dev
 
@@ -97,18 +74,13 @@
 # move application files to target locations
 
 if [ $install -eq 1 ]; then
-   mkdir -p "$DATADIR" "$DOCDIR" "$BINDIR"
-   chmod 0755 "$DATADIR" "$DOCDIR" "$BINDIR"
+   mkdir -p -m o=rx "$RPM_BUILD_ROOT$DATADIR" "$RPM_BUILD_ROOT$BINDIR"
    if [ $? -ne 0 ]; then exit 1; fi
-   cp -f  *.pdf  "$DOCDIR"
-   cp -f  README  "$DOCDIR"
-   cp -f  COPYING  "$DOCDIR"
-   cp -f  changelog  "$DOCDIR"
-   cp -f -r  icons  "$DATADIR"
-   cp -f  *.xtext  "$DATADIR"
-   cp -f  parameter*  "$DATADIR" > /dev/null 2>&1
-   cp -f  *.conf  "$DATADIR" > /dev/null 2>&1
-   cp -f  $APPNAME  "$BINDIR"
+   cp -f -r  icons  "$RPM_BUILD_ROOT$DATADIR"
+   cp -f  *.xtext  "$RPM_BUILD_ROOT$DATADIR"
+   cp -f  parameter*  "$RPM_BUILD_ROOT$DATADIR" > /dev/null 2>&1
+   cp -f  *.conf  "$RPM_BUILD_ROOT$DATADIR" > /dev/null 2>&1
+   cp -f  $APPNAME  "$RPM_BUILD_ROOT$BINDIR"
    if [ $? -ne 0 ]; then exit 1; fi
    echo "$APPNAME installation successful"
 fi
