--- apps/ascompose.c	2008-06-24 11:46:11.000000000 -0500
+++ apps/ascompose.c	2009-08-26 09:20:13.000000000 -0500
@@ -43,13 +43,19 @@
 # endif
 #endif
 
-#if defined(SHAPE) && !defined(X_DISPLAY_MISSING)
-#include <X11/extensions/shape.h>
-#endif /* SHAPE */
+
 #include "../afterbase.h"
 #include "../afterimage.h"
 #include "common.h"
 
+#if !defined(X_DISPLAY_MISSING)
+int asvisual_empty_XErrorHandler (Display * dpy, XErrorEvent * event);
+
+# if defined(SHAPE)
+#  include <X11/extensions/shape.h>
+# endif /* SHAPE */
+#endif /* X_DISPLAY_MISSING */
+
 
 
 /****h* libAfterImage/ascompose
@@ -274,6 +280,7 @@
 Window make_main_window(Bool on_root, ASComposeWinProps *props);	
 
 int screen = 0, depth = 0;
+Display *dpy = NULL;
 
 int main(int argc, char** argv) {
 
@@ -731,7 +738,10 @@
 	if (main_window_props.canvas)
 		XFreePixmap(dpy, main_window_props.canvas); 
 	if( dpy )
+	{
         XCloseDisplay (dpy);
+		dpy = NULL;
+	}
 #endif
 	LOCAL_DEBUG_OUT( "display Closed%s","");
 #ifdef DEBUG_ALLOCS
@@ -793,8 +803,6 @@
 	return w;
 }
 
-int asvisual_empty_XErrorHandler (Display * dpy, XErrorEvent * event);
-
 Bool
 set_root_pixmap_property(long pmap) /* Must have long type to work with XChangeProp on 64 bit machines !!!*/
 {
@@ -819,13 +827,13 @@
 		for (i = 0 ; i < 2 ; ++i)
 			if (atoms[i])
 			{
-				LOCAL_DEBUG_OUT("atoms[%d] = %X", i, atoms[i]);
+				LOCAL_DEBUG_OUT("atoms[%d] = %lX", i, atoms[i]);
 				data.long_ptr = NULL ;
 		        XGetWindowProperty(dpy, root, atoms[i], 0L, 1L, False, AnyPropertyType, &type, &format, &nitems, &after, &data.uc_ptr);
 				if (data.long_ptr == NULL)
 					break;
 				pmaps[i] = data.long_ptr[0] ;
-				LOCAL_DEBUG_OUT("pmaps[%d] = %X", i, pmaps[i]);
+				LOCAL_DEBUG_OUT("pmaps[%d] = %lX", i, pmaps[i]);
 				if (format != 32 || nitems == 0 || pmaps[i] != pmaps[0] || type != XA_PIXMAP) 
 					break;
 			}
@@ -833,7 +841,7 @@
 		{
 			/* XKillClient is a dangerous affair since properties may hold stale values tha are no longer valid */
 			int (*oldXErrorHandler) (Display *, XErrorEvent *) = XSetErrorHandler (asvisual_empty_XErrorHandler);
-			LOCAL_DEBUG_OUT("killing client for pmap %X", pmaps[0]);
+			LOCAL_DEBUG_OUT("killing client for pmap %lX", pmaps[0]);
            	XKillClient(dpy, pmaps[0]);
 			XSync(dpy, False);
 			XSetErrorHandler (oldXErrorHandler);
@@ -844,7 +852,7 @@
 		if (!atoms[i])
 			if ( (atoms[i] = XInternAtom (dpy, names[i], False)) == None)
 				break;
-		LOCAL_DEBUG_OUT("Changing property %X to pmap id %X", atoms[i], pmap);
+		LOCAL_DEBUG_OUT("Changing property %lX to pmap id %lX", atoms[i], pmap);
     	XChangeProperty(dpy, root, atoms[i], XA_PIXMAP, 32, PropModeReplace, (unsigned char *) &pmap, 1);
 	}
 	if (i >= 2)
