--- Temperature.cc	2008-06-30 14:46:07.000000000 -0500
+++ Temperature.cc	2008-06-30 15:08:19.000000000 -0500
@@ -28,6 +28,7 @@
 #include <cstdlib>
 #include <cstdio>
 #include <string>
+#include <string.h>
 #include <unistd.h>
 #include <csignal>
 #include <cerrno>
--- Xpm.h	2002-04-05 17:44:30.000000000 -0600
+++ Xpm.h	2008-06-30 16:04:10.000000000 -0500
@@ -24,6 +24,7 @@
 
 #include <X11/Xlib.h>
 #include <X11/xpm.h>
+#include <string.h>
 
 class Xpm 
 {
