--- Makefile	2008-07-03 00:34:57.000000000 -0500
+++ Makefile	2010-08-23 23:28:38.000000000 -0500
@@ -1,10 +1,7 @@
 # Makefile
 
 # Installation directory
-PREFIX=/usr/local
-
-# Use Xft for the menu
-USE_XFT=yes
+PREFIX=/usr
 
 VERSION=0.9
 
@@ -19,19 +16,9 @@
 OS=$(shell uname -s)
 #For Linux:
 ifeq ($(OS),Linux)
-LIBS=-L/usr/X11R6/lib -lX11 -lXext -lXpm
+LIBS=-L/usr/X11R6/lib -L/usr/lib -L/usr/lib64 -lX11 -lXext -lXpm -lXft
 OBJS += cpu_linux.o
 endif
-#For FreeBSD:
-ifeq ($(OS),FreeBSD)
-LIBS=-L/usr/X11R6/lib -lX11 -lXext -lXpm -lkvm
-OBJS += cpu_freebsd.o
-endif
-
-ifeq ($(USE_XFT),yes)
-CFLAGS += -DUSE_XFT $(shell xft-config --cflags)
-LIBS += $(shell xft-config --libs)
-endif
 
 CC=gcc
 RM=rm -rf
