--- Makefile	2012-02-27 20:18:57.000000000 -0600
+++ Makefile	2012-02-27 21:12:33.000000000 -0600
@@ -1,7 +1,7 @@
 # Makefile
 
 # Installation directory
-PREFIX=/usr/local
+PREFIX=/usr
 
 # Use Xft for the menu
 USE_XFT=yes
@@ -12,13 +12,14 @@
 BINDIR=$(PREFIX)/bin
 MANPREFIX=$(PREFIX)/share/man
 MANDIR=$(MANPREFIX)/man1
+DESTDIR=
 MANUAL=$(PROG).1
 OBJS=main.o menu.o
 CFLAGS=-O2 -Wall -DVERSION=\"$(VERSION)\"
 
 OS=$(shell uname -s)
 ifeq ($(OS),Linux)
-LIBS=-L/usr/X11R6/lib -lX11 -lXext -lXpm
+LIBS=-I/usr/include/freetype2 -L/usr/lib -L/usr/lib64 -lX11 -lXext -lXpm
 OBJS+=mem_linux.o
 endif
 
@@ -28,8 +29,8 @@
 endif
 
 ifeq ($(USE_XFT),yes)
-CFLAGS += -DUSE_XFT $(shell xft-config --cflags)
-LIBS += $(shell xft-config --libs)
+CFLAGS += -DUSE_XFT -I/usr/include/freetype2
+LIBS += -lXft
 endif
 
 CC=gcc
@@ -46,8 +47,10 @@
 clean:
 	$(RM) *.o $(PROG) *~ *.bak *.BAK .xvpics
 install: $(PROG) $(MANUAL)
-	$(INST) -m 755 $(PROG) $(BINDIR)
-	$(INST) -m 644 $(MANUAL) $(MANDIR)
+	mkdir -p $(DESTDIR)$(BINDIR)
+	$(INST) -m 755 $(PROG) $(DESTDIR)$(BINDIR)
+	mkdir -p $(DESTDIR)$(MANDIR)
+	$(INST) -m 644 $(MANUAL) $(DESTDIR)$(MANDIR)
 uninstall:
 	$(RM) $(BINDIR)/$(PROG)
 	$(RM) $(MANDIR)/$(MANUAL)
