--- Makefile	2005-03-14 01:28:31.000000000 -0600
+++ Makefile	2012-02-13 23:45:01.000000000 -0600
@@ -14,8 +14,10 @@
 CC = gcc
 LD = gcc
 COPTS = -Wall -O2 -D'VERSION="$(VERSION)"' -D'NAME="$(NAME)"'
-LDOPTS = -lX11 -L/usr/X11R6/lib -lXpm -lXext
-BINDIR = /usr/local/bin
+LDOPTS = -lX11 -L/usr/lib -L/usr/lib64 -lXpm -lXext
+PREFIX = /usr
+BINDIR = $(PREFIX)/bin
+DESTDIR=
 
 BIN =	wmifinfo
 FILES = wmifinfo.o xutils.o 
@@ -37,7 +39,8 @@
 	rm -f *.o $(BIN) core ./.#* *.orig *.rej
 
 install:
-	cp $(BIN) $(BINDIR)
+	mkdir -p $(DESTDIR)$(BINDIR)
+	install -m 755 $(BIN) $(DESTDIR)$(BINDIR)
 
 dist:	clean
 	rm -rf /tmp/wmifinfo-$(VERSION)
