--- wminfo/configure	2013-04-08 14:07:20.000000000 -0500
+++ wminfo/configure	2013-07-10 18:39:06.428976736 -0500
@@ -28,7 +28,7 @@
 fi
 if [ "$PREFIX" == "" ]
 then
-    PREFIX=/usr/local
+    PREFIX=/usr
 fi
 
 FILE=Makefile
@@ -36,19 +36,20 @@
 
 cat >> $FILE <<EOF
 PREFIX=$PREFIX
-PACKAGE=wminfo-$version
+PACKAGE=wminfo
 
 BINDIR=\$(PREFIX)/bin
-DOCDIR=\$(PREFIX)/doc/\$(PACKAGE)
-MANDIR=\$(PREFIX)/man/man1
+DOCDIR=\$(PREFIX)/share/doc/\$(PACKAGE)
+MANDIR=\$(PREFIX)/share/man/man1
 
-LIBDIR = -L/usr/X11R6/lib
+LIBDIR = -L/usr/lib -L/usr/lib64
 LIBS   = -lXpm -lXext -lX11
+DESTDIR=
 CFLAGS = -march=native -O2 -pipe
 OBJS   = wminfo.o ../wmgeneral/wmgeneral.o ../wmgeneral/misc.o ../wmgeneral/list.o
 
 .c.o:
-	gcc -I/usr/X11R6/share/include \$(CFLAGS) -c -Wall \$< -o \$*.o
+	gcc -I/usr/include \$(CFLAGS) -c -Wall \$< -o \$*.o
 
 all: \$(OBJS)
 	gcc \$(CFLAGS) -o wminfo \$^ -lXext \$(LIBDIR) \$(LIBS) && strip --strip-unneeded wminfo
@@ -75,34 +76,33 @@
 	rm -f Makefile
 
 install:
-	cp -p wminfo \$(BINDIR)
-	chmod 755 \$(BINDIR)/wminfo
-	chown root:root \$(BINDIR)/wminfo
-
-	cp -p regex \$(BINDIR)
-	chmod 755 \$(BINDIR)/regex
-	chown root:root \$(BINDIR)/regex
+	mkdir -p \$(DESTDIR)\$(BINDIR)
+	cp -p wminfo \$(DESTDIR)\$(BINDIR)
+	chmod 755 \$(DESTDIR)\$(BINDIR)/wminfo
+
+	cp -p regex \$(DESTDIR)\$(BINDIR)
+	chmod 755 \$(DESTDIR)\$(BINDIR)/regex
 
 	for f in \$(CONTRIB) \\
-	; do cp -p ../contrib/\$\$f \$(BINDIR) ; done
+	; do cp -p ../contrib/\$\$f \$(DESTDIR)\$(BINDIR) ; done
 
-	if [ ! -d \$(MANDIR) ]; then mkdir -p \$(MANDIR); fi
+	mkdir -p \$(DESTDIR)\$(MANDIR)
 
-	cp -p ../man/wminfo.1 \$(MANDIR)
+	cp -p ../man/wminfo.1 \$(DESTDIR)\$(MANDIR)
 
-	if [ ! -d \$(DOCDIR) ]; then mkdir -p \$(DOCDIR); fi
+	mkdir -p \$(DESTDIR)\$(DOCDIR)
 
 	for f in \$(DOCS) \\
-	; do cp -p ../\$\$f \$(DOCDIR) ; done
+	; do cp -p ../\$\$f \$(DESTDIR)\$(DOCDIR) ; done
 
 	for d in contrib \$(DIRS) \\
-	; do if [ ! -d \$(DOCDIR)/\$\$d ]; then mkdir \$(DOCDIR)/\$\$d; fi ; \\
-	cp -p ../\$\$d/* \$(DOCDIR)/\$\$d ; done
+	; do if [ ! -d \$(DESTDIR)\$(DOCDIR)/\$\$d ]; then mkdir \$(DESTDIR)\$(DOCDIR)/\$\$d; fi ; \\
+	cp -p ../\$\$d/* \$(DESTDIR)\$(DOCDIR)/\$\$d ; done
 
 	for d in \$(DIRS) \\
-	; do if [ ! -x \$(DOCDIR)/\$\$d/run-all-wminfo-plugins ]; then ln -s \$(BINDIR)/run-all-wminfo-plugins \$(DOCDIR)/\$\$d/run-all-wminfo-plugins; fi ; done
+	; do if [ ! -x \$(DOCDIR)/\$\$d/run-all-wminfo-plugins ]; then ln -s \$(BINDIR)/run-all-wminfo-plugins \$(DESTDIR)\$(DOCDIR)/\$\$d/run-all-wminfo-plugins; fi ; done
 
-	if [ ! -x \$(BINDIR)/\$\$d/empty.wmi ]; then ln -s \$(BINDIR)/bash-empty.wmi \$(BINDIR)/\$\$d/empty.wmi; fi
+	if [ ! -x \$(DESTDIR)\$(BINDIR)/\$\$d/empty.wmi ]; then ln -s \$(BINDIR)/bash-empty.wmi \$(DESTDIR)\$(BINDIR)/\$\$d/empty.wmi; fi
 
 uninstall:
 	for f in wminfo regex empty.wmi \$(CONTRIB) \\
