--- wminfo/configure	2011-10-30 09:00:56.000000000 -0500
+++ wminfo/configure	2011-11-29 21:47:40.000000000 -0600
@@ -28,22 +28,23 @@
 fi
 if [ "$PREFIX" == "" ]
 then
-    PREFIX=/usr/local
+    PREFIX=/usr
 fi
 
 FILE=Makefile
 
 > $FILE
 echo PREFIX=$PREFIX >> $FILE
-echo PACKAGE=wminfo-$version >> $FILE
+echo PACKAGE=wminfo >> $FILE
 echo >> $FILE
 echo BINDIR=\$\(PREFIX\)/bin >> $FILE
-echo DOCDIR=\$\(PREFIX\)/doc/\$\(PACKAGE\) >> $FILE
-echo MANDIR=\$\(PREFIX\)/man/man1 >> $FILE
+echo DOCDIR=\$\(PREFIX\)/share/doc/\$\(PACKAGE\) >> $FILE
+echo MANDIR=\$\(PREFIX\)/share/man/man1 >> $FILE
 echo >> $FILE
 cat >> $FILE <<EOF
-LIBDIR = -L/usr/X11R6/lib
+LIBDIR = -L/usr/lib -L/usr/lib64
 LIBS   = -lXpm -lXext -lX11
+DESTDIR=
 FLAGS = -O2
 OBJS =	wminfo.o \\
 		../wmgeneral/wmgeneral.o \\
@@ -51,7 +52,7 @@
 		../wmgeneral/list.o
 
 .c.o:
-	cc -I/usr/X11R6/share/include \$(FLAGS) -c -Wall \$< -o \$*.o
+	cc -I/usr/include \$(FLAGS) -c -Wall \$< -o \$*.o
 
 wminfo: \$(OBJS)
 	cc \$(FLAGS) -o wminfo \$^ -lXext \$(LIBDIR) \$(LIBS)
@@ -68,24 +69,23 @@
 	rm -f Makefile
 
 install: wminfo
-	cp -p wminfo \$(BINDIR)/
-	chmod 755 \$(BINDIR)/wminfo
-	chown root:root \$(BINDIR)/wminfo
-	cp -p ../contrib/cp1250-up2low ../contrib/cp1251-up2low ../contrib/default.wmi ../contrib/empty.wmi ../contrib/html-iso1 ../contrib/iso1-up2low ../contrib/iso2-up2low ../contrib/iso5-up2low ../contrib/koi8r-up2low ../contrib/plugin ../contrib/punctuation ../contrib/suspend ../contrib/up2low ../contrib/utf8-iso1 ../contrib/utf8-iso2 ../contrib/utf8-iso5 ../contrib/utf8-up2low \$(BINDIR)/
-	if [ ! -d \$(MANDIR) ]; then mkdir -p \$(MANDIR); fi
-	cp -p ../doc/wminfo.1 \$(MANDIR)
-	if [ ! -d \$(DOCDIR) ]; then mkdir \$(DOCDIR); fi
-	cp -p ../COPYING ../ChangeLog ../INSTALL ../Plugins-HOWTO ../README ../README.Privacy ../TODO \$(DOCDIR)
-	if [ ! -d \$(DOCDIR)/contrib ]; then mkdir \$(DOCDIR)/contrib; fi
-	cp -p ../contrib/* \$(DOCDIR)/contrib
-	if [ ! -d \$(DOCDIR)/plugins.offline ]; then mkdir \$(DOCDIR)/plugins.offline; fi
-	cp -p ../plugins.offline/* \$(DOCDIR)/plugins.offline
-	if [ ! -d \$(DOCDIR)/plugins.online ]; then mkdir \$(DOCDIR)/plugins.online; fi
-	cp -p ../plugins.online/* \$(DOCDIR)/plugins.online
-	if [ ! -d \$(DOCDIR)/plugins.system ]; then mkdir \$(DOCDIR)/plugins.system; fi
-	cp -p ../plugins.system/* \$(DOCDIR)/plugins.system
-	if [ ! -d \$(DOCDIR)/samples ]; then mkdir \$(DOCDIR)/samples; fi
-	cp -p ../samples/* \$(DOCDIR)/samples
+	mkdir -p \$(DESTDIR)\$(BINDIR)
+	cp -p wminfo \$(DESTDIR)\$(BINDIR)/
+	chmod 755 \$(DESTDIR)\$(BINDIR)/wminfo
+	cp -p ../contrib/cp1250-up2low ../contrib/cp1251-up2low ../contrib/default.wmi ../contrib/empty.wmi ../contrib/html-iso1 ../contrib/iso1-up2low ../contrib/iso2-up2low ../contrib/iso5-up2low ../contrib/koi8r-up2low ../contrib/plugin ../contrib/punctuation ../contrib/suspend ../contrib/up2low ../contrib/utf8-iso1 ../contrib/utf8-iso2 ../contrib/utf8-iso5 ../contrib/utf8-up2low \$(DESTDIR)\$(BINDIR)/
+	mkdir -p \$(DESTDIR)\$(MANDIR)
+	cp -p ../doc/wminfo.1 \$(DESTDIR)\$(MANDIR)
+	mkdir -p \$(DESTDIR)\$(DOCDIR)
+	mkdir -p \$(DESTDIR)\$(DOCDIR)/contrib
+	cp -p ../contrib/* \$(DESTDIR)\$(DOCDIR)/contrib
+	mkdir -p \$(DESTDIR)\$(DOCDIR)/plugins.offline
+	cp -p ../plugins.offline/* \$(DESTDIR)\$(DOCDIR)/plugins.offline
+	mkdir -p \$(DESTDIR)\$(DOCDIR)/plugins.online
+	cp -p ../plugins.online/* \$(DESTDIR)\$(DOCDIR)/plugins.online
+	mkdir -p \$(DESTDIR)\$(DOCDIR)/plugins.system
+	cp -p ../plugins.system/* \$(DESTDIR)\$(DOCDIR)/plugins.system
+	mkdir -p \$(DESTDIR)\$(DOCDIR)/samples
+	cp -p ../samples/* \$(DESTDIR)\$(DOCDIR)/samples
 
 uninstall:
 	rm -f \$(BINDIR)/wminfo
