--- wmsystray/Makefile	2004-03-01 10:15:26.000000000 +0100
+++ wmsystray/Makefile	2004-03-01 10:15:37.000000000 +0100
@@ -1,7 +1,8 @@
 include ../Rules.make
 
 CC = gcc
-CFLAGS = ${X11CFLAGS} -DTRACE_LEVEL=${TRACE_LEVEL} -I../xembed -Wall -g
+OPTFLAGS = -g
+CFLAGS = ${X11CFLAGS} -DTRACE_LEVEL=${TRACE_LEVEL} -I../xembed -Wall $(OPTFLAGS)
 LDFLAGS = ${X11LDFLAGS} -lXpm
 
 OBJS = main.o ui.o systray.o xpms.o
--- xembed/Makefile	2004-03-01 10:15:43.000000000 +0100
+++ xembed/Makefile	2004-03-01 10:15:58.000000000 +0100
@@ -1,6 +1,7 @@
 include ../Rules.make
 
-CFLAGS = -DTRACE_LEVEL=${TRACE_LEVEL} ${X11CFLAGS} -Wall -g
+OPTFLAGS = -g
+CFLAGS = -DTRACE_LEVEL=${TRACE_LEVEL} ${X11CFLAGS} -Wall $(OPTFLAGS)
 LDFLAGS = ${X11LDFLAGS}
 
 all: xembed.o
