--- wmtz/Makefile	2011-09-05 18:46:58.000000000 -0500
+++ wmtz/Makefile	2012-02-13 09:49:21.000000000 -0600
@@ -1,8 +1,13 @@
 CC = gcc
-LIBDIR = -L/usr/X11R6/lib
+PREFIX = /usr
+BINDIR = $(PREFIX)/bin
+LIBPATH = $(PREFIX)/lib
+MANDIR = $(PREFIX)/share/man
+SYSCONFDIR = /etc
+LIBDIR = -L$(LIBPATH)
 LIBS   = -lXpm -lXext -lX11 -lm
 FLAGS = -O2
-DESTDIR = /usr/local
+DESTDIR =
 OBJS =	wmtz.o \
 		../wmgeneral/wmgeneral.o \
 		../wmgeneral/misc.o \
@@ -10,7 +15,7 @@
 
 
 .c.o:
-	$(CC) -I/usr/X11R6/include $(FLAGS) -c -Wall $< -o $*.o
+	$(CC) -I/usr/include $(FLAGS) -c -Wall $< -o $*.o
 
 wmtz: $(OBJS)
 	$(CC) $(FLAGS) -o wmtz $^ -lXext $(LIBDIR) $(LIBS)
@@ -25,18 +30,12 @@
 	rm -f *~
 
 install:: wmtz
-	cp -f wmtz $(DESTDIR)/bin/
-	chmod 755 $(DESTDIR)/bin/wmtz
-	chown root:root $(DESTDIR)/bin/wmtz
-	mkdir -p $(DESTDIR)/man/man1
-	cp -f wmtz.1 $(DESTDIR)/man/man1/
-	chmod 644 $(DESTDIR)/man/man1/wmtz.1
-	chown root:root $(DESTDIR)/man/man1/wmtz.1
-	cp -f wmtzrc $(HOME)/.wmtzrc
-	chmod 600 $(HOME)/.wmtzrc
-	cp -f wmtzrc /etc/wmtzrc
-	chmod 644 /etc/wmtzrc
-	chown root:root /etc/wmtzrc
+	mkdir -p $(DESTDIR)/$(BINDIR)
+	install -m 755 wmtz $(DESTDIR)$(BINDIR)
+	mkdir -p $(DESTDIR)$(MANDIR)/man1
+	install -m 644 wmtz.1 $(DESTDIR)$(MANDIR)/man1/
+	mkdir -p $(DESTDIR)/$(SYSCONFDIR)
+	install -m 644 wmtzrc $(DESTDIR)$(SYSCONFDIR)/wmtzrc
 	@echo "wmtz: Installation finished..."
 
 uninstall:: 
