--- wmweather+-2.9/Makefile.in.old	2005-02-26 16:36:46.306281104 -0500
+++ wmweather+-2.9/Makefile.in	2005-02-26 16:37:16.137746032 -0500
@@ -66,7 +66,7 @@
 wmweather__OBJECTS = $(am_wmweather__OBJECTS)
 wmweather__DEPENDENCIES = wmgeneral/libwmgeneral.a \
 	wmgeneral/libwmgeneral-x11.a b0rken/libb0rken.a
-DEFAULT_INCLUDES = -I. -I$(srcdir) -I.
+DEFAULT_INCLUDES = -I. -I$(srcdir) -I. -I/usr/include/pcre 
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 @AMDEP_TRUE@DEP_FILES = ./$(DEPDIR)/animation.Po ./$(DEPDIR)/avn.Po \
